velox/common/encode/Base64.cpp

```diff
@@ -433,6 +433,15 @@ std::string Base64::encode_url(const folly::IOBuf* data) {
   return encodeImpl(IOBufWrapper(data), kBase64UrlCharset, false);
 }
 
+void Base64::decodeUrl(
+    const char* src,
+    size_t src_len,
+    char* dst,
+    size_t dst_len,
+    bool hasPad) {
+  decodeImpl(src, src_len, dst, dst_len, kBase64UrlReverseIndexTable, hasPad);
```

Comment:
The existing `decode_url` on line 445 estimates and resizes the output. Why is this implementation different?

Reply:
the line 454 function signature has padding as false hardcoded, however, if you look at the rfc 4648 sec-5, it says, padding is optional, meaning - there could be padding or may not be padding present in the encoded input string, thus have to come up with new decodeUrl signature.

`[RFC 4648 §5](https://datatracker.ietf.org/doc/html/rfc4648#section-5): base64url (URL- and filename-safe standard)[[a]](https://en.wikipedia.org/wiki/Base64#cite_note-common-7)`

Reply:
Does this mean the existing implementation has a bug? We can probably fix that in a follow-up PR by reusing this logic.

Reply:
Based on the RFC, seems like there is a bug in the existing implementation as optional padding is not taken into account. Sure, will open a new PR to get that fixed.